README.md

```diff
@@ -23,8 +23,95 @@ Terra Toolkit is a utility module used to facilitate independent development of
+const webpackConfig = require('./webpack.config');
+const getPackageTestDirectories = require('terra-toolkit/lib/nightwatch/setup-helper.js').getPackageTestDirectories;
+
+let srcFolders;
+const isRepoTest = !process.cwd().includes('/packages/');
+
+if (isRepoTest) {
+  srcFolders = getPackageTestDirectories('lerna.json');
+} else {
+  srcFolders = 'tests/nightwatch/';
+}
+
+const config = nightwatchConfig(webpackConfig, srcFolders, 9000);
+
+module.exports = config;
+```
+
+
+### Writing Tests
+
+There are a few things to note about the nightwatch configuration generated by Terra-Toolkit:
+
+- Test files should use `*-spec.js` naming format. The generated configuration filters on `**/*-spec.js`.
+- An after test hook is provided which calls the nightwatch `.end()` function.
+- Use `${browser.launchUrl}/#/test_url_path` to direct test urls. The generated configuration sets the launchUrl to `http://localhost:${port}`.
+- [ChromeDriver](https://www.npmjs.com/package/chromedriver) is the default testing environment and it's default browser size is 800x600.
+
+To assist with testing various browser sizes, Terra-toolkit provides the responsive test helpers, `resizeTo` and `screenWidth`.
+- `resizeTo` wraps the nightwatch test suite and handles browser resizing on the breakpoints provided. The breakpoint options, `tiny`, `small`, `medium`, `large`, `huge`, and `enormous`, represent the screen widths tested; these correspond to the screen widths of 470px, 622px, 838px, 1000px, 1300px, and 1500px respectively.
+- `screenWidth` returns the current screen width. This helper must be used in conjunction with `resizeTo`.
+
+```javascript
+const { resizeTo, screenWidth } = require('terra-tookit/lib/nightwatch/responsive-helpers');
+
+module.exports = resizeTo(['tiny', 'small', 'medium', 'large', 'huge', 'enormous'], {
+  'Static test path one renders correctly': (browser) => {
+    browser.url(`${browser.launchUrl}/#/test-path-1`);
+    browser.expect.element('.test').text.to.equal('Test');
+  },
+
+  'Responsive test path two renders correctly': (browser) => {
+    browser.url(`${browser.launchUrl}/#/test-path-2`);
+    const width = screenWidth(browser);
+
+    if (width <= browser.globals.breakpoints.tiny[0]) {
+      browser.expect.element('.test').text.to.equal('Tiny Response');
+    } else {
+      browser.expect.element('.test').text.to.equal('Default Response');
+    }
+  },
+});
+```
+
+Finally, it is **highly recommended** to use the expect API over the assert API. Expect is a Behavior Driven Development interface which aligns more appropriately with react renderings and interactions.
+
+#### Writing Tests with ChromeDriver:
+ - ChromeDriver clicks the center of an element.
+ - ChromeDriver cannot click moving objects. More information [here](https://sites.google.com/a/chromium.org/chromedriver/help/clicking-issues).
+ - ChromeDriver's clearValue functions correctly, but does not trigger a change for an element.
```

Comment:
Is there more info on this clearValue behavior?

Reply:
No not really. I spent a lot of time trying to update this terra-date-picker [test](https://github.com/cerner/terra-core/blob/master/packages/terra-date-picker/tests/nightwatch/date-picker-spec.js#L79). 

Upon evaluation of the input field, `browser.clearValue( )` would appropriate clear the value of the input and it would clear the text of the input, yet no onChange was trigger by the component. I did a bunch of searching and I ultimately determined this as an issue of ChromeDriver since this test functioned correctly in PhantomJS.